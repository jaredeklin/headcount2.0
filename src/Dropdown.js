import React, { Component } from 'react';
import './styles/dropdown.css';
import { dataList, getProgram } from './dataCategories';

class Dropdown extends Component  {

  handleChange = (event) => {
    const newDataSet = getProgram(event.target.value);
    
    this.props.getStats(newDataSet);
  }
  
  render() {
    return (
      <div className='dropdown-container'>
        <h3>Choose a category to compare districts:</h3>
        <select onChange={this.handleChange}>
          {
            dataList.map((dataSet, index) => <option key={index}>{dataSet.type}</option>)
          }
        </select>
      </div>
    )
  }
}

export default Dropdown