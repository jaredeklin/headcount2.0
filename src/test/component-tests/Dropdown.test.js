import React from 'react';
import { shallow, mount } from 'enzyme';
import Dropdown from '../../Dropdown.js';

describe('Dropdown', () => {

  it('should match the snapshot', () => {
    const wrapper = shallow(<Dropdown />);
    expect(wrapper).toMatchSnapshot();
  })

  it('should call getStats in handleChange', () => {
    const mockGetStats = jest.fn();
    const wrapper = shallow(<Dropdown getStats={mockGetStats} />);
    const mockEvent = {target: {value: 'somestring'}}

    wrapper.instance().handleChange(mockEvent);
    expect(mockGetStats).toHaveBeenCalled()
  })
})