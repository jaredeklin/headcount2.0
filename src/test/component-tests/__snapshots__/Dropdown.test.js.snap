// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dropdown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Choose a category to compare districts:
</h3>,
        <select
          onChange={[Function]}
>
          <option>
                    Kindergarteners in Full Day Programs
          </option>
          <option>
                    High School Graduation Rates
          </option>
          <option>
                    Median Household Income
          </option>
          <option>
                    Special Education
          </option>
          <option>
                    Title I Students
          </option>
          <option>
                    Remediation In Higher Education
          </option>
          <option>
                    Online Pupil Enrollment
          </option>
          <option>
                    Pupil Enrollment
          </option>
</select>,
      ],
      "className": "dropdown-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Choose a category to compare districts:",
        },
        "ref": null,
        "rendered": "Choose a category to compare districts:",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option>
              Kindergarteners in Full Day Programs
</option>,
            <option>
              High School Graduation Rates
</option>,
            <option>
              Median Household Income
</option>,
            <option>
              Special Education
</option>,
            <option>
              Title I Students
</option>,
            <option>
              Remediation In Higher Education
</option>,
            <option>
              Online Pupil Enrollment
</option>,
            <option>
              Pupil Enrollment
</option>,
          ],
          "onChange": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": "Kindergarteners in Full Day Programs",
            },
            "ref": null,
            "rendered": "Kindergarteners in Full Day Programs",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "High School Graduation Rates",
            },
            "ref": null,
            "rendered": "High School Graduation Rates",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": "Median Household Income",
            },
            "ref": null,
            "rendered": "Median Household Income",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "children": "Special Education",
            },
            "ref": null,
            "rendered": "Special Education",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "4",
            "nodeType": "host",
            "props": Object {
              "children": "Title I Students",
            },
            "ref": null,
            "rendered": "Title I Students",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "5",
            "nodeType": "host",
            "props": Object {
              "children": "Remediation In Higher Education",
            },
            "ref": null,
            "rendered": "Remediation In Higher Education",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "6",
            "nodeType": "host",
            "props": Object {
              "children": "Online Pupil Enrollment",
            },
            "ref": null,
            "rendered": "Online Pupil Enrollment",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "7",
            "nodeType": "host",
            "props": Object {
              "children": "Pupil Enrollment",
            },
            "ref": null,
            "rendered": "Pupil Enrollment",
            "type": "option",
          },
        ],
        "type": "select",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Choose a category to compare districts:
</h3>,
          <select
            onChange={[Function]}
>
            <option>
                        Kindergarteners in Full Day Programs
            </option>
            <option>
                        High School Graduation Rates
            </option>
            <option>
                        Median Household Income
            </option>
            <option>
                        Special Education
            </option>
            <option>
                        Title I Students
            </option>
            <option>
                        Remediation In Higher Education
            </option>
            <option>
                        Online Pupil Enrollment
            </option>
            <option>
                        Pupil Enrollment
            </option>
</select>,
        ],
        "className": "dropdown-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Choose a category to compare districts:",
          },
          "ref": null,
          "rendered": "Choose a category to compare districts:",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option>
                Kindergarteners in Full Day Programs
</option>,
              <option>
                High School Graduation Rates
</option>,
              <option>
                Median Household Income
</option>,
              <option>
                Special Education
</option>,
              <option>
                Title I Students
</option>,
              <option>
                Remediation In Higher Education
</option>,
              <option>
                Online Pupil Enrollment
</option>,
              <option>
                Pupil Enrollment
</option>,
            ],
            "onChange": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": "Kindergarteners in Full Day Programs",
              },
              "ref": null,
              "rendered": "Kindergarteners in Full Day Programs",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": "High School Graduation Rates",
              },
              "ref": null,
              "rendered": "High School Graduation Rates",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": "Median Household Income",
              },
              "ref": null,
              "rendered": "Median Household Income",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "3",
              "nodeType": "host",
              "props": Object {
                "children": "Special Education",
              },
              "ref": null,
              "rendered": "Special Education",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "4",
              "nodeType": "host",
              "props": Object {
                "children": "Title I Students",
              },
              "ref": null,
              "rendered": "Title I Students",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "5",
              "nodeType": "host",
              "props": Object {
                "children": "Remediation In Higher Education",
              },
              "ref": null,
              "rendered": "Remediation In Higher Education",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "6",
              "nodeType": "host",
              "props": Object {
                "children": "Online Pupil Enrollment",
              },
              "ref": null,
              "rendered": "Online Pupil Enrollment",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "7",
              "nodeType": "host",
              "props": Object {
                "children": "Pupil Enrollment",
              },
              "ref": null,
              "rendered": "Pupil Enrollment",
              "type": "option",
            },
          ],
          "type": "select",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
